packages/cli/src/lib/config/command-object.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { dirname, join } from 'path';
+import { fileURLToPath } from 'url';
+import { objectToCliArgs } from '@code-pushup/utils';
+import { middlewares } from '../middlewares';
+import { options } from '../options';
+import { yargsCli } from '../yargs-cli';
+import { yargsConfigCommandObject } from './command-object';
+
+const baseArgs = [
+  ...objectToCliArgs({
+    verbose: true,
+    config: join(
+      fileURLToPath(dirname(import.meta.url)),
+      '..',
+      '..',
+      '..',
+      'test',
+      'config.mock.ts',
+    ),
+  }),
+];
+const cli = (args: string[]) =>
+  yargsCli(['config', ...args], {
+    options,
+    middlewares,
+    commands: [yargsConfigCommandObject()],
+  });
+
+describe('config-command-object', () => {
+  it('should override config with CLI arguments', async () => {
+    const args = [
+      ...baseArgs,
+      ...objectToCliArgs({
+        format: 'md',
+      }),
+    ];
+    const parsedArgv = await cli(args).parseAsync();
+    expect(parsedArgv.persist.outputDir).toBe('tmp');
+    expect(parsedArgv.persist?.format).toEqual(['md']);
```

Comment:
This is only testing the arguments, but what's logged to the console isn't tested at all, even though that's the end result we care about.

Reply:
added snapshot to stdout

Reply:
mocked console, added snapshot. Only problem I have ATM is getting rid of the path in the snapshot.

Reply:
this is not working:

```typescript
 beforeEach(async () => {
    logs = [];
    mockConsole(msg => {
      const cleanMsg = msg.replace(
        /(\\"config\\\\": \\\\".*?\\\\")/m,
        `\\"config\\": \\"XXX\\"`,
      );
      logs.push(cleanMsg);
    });
  });
```